src/hotspot/cpu/riscv/nativeInst_riscv.cpp

```diff
@@ -89,22 +89,47 @@ void NativeCall::print() {
   tty->print_cr(PTR_FORMAT ": auipc,ld,jalr x1, offset/reg, ", p2i(instruction_address()));
 }
 
+void NativeCall::optimize_call(address dest, bool mt_safe) {
+  // Skip over auipc + ld
+  address jal_pc = instruction_address() + 2 * NativeInstruction::instruction_size;
+  uint32_t *jal_pos = (uint32_t *)jal_pc;
```

Comment:
Is it possible to lose some data in this conversion?
If not, maybe an assert here?